src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -586,7 +586,10 @@ public void TrackRequest(RequestTelemetry request)
         /// </remarks>
         public void Flush()
         {
-            this.configuration.TelemetryChannel.Flush();
+            if (this.configuration.TelemetryChannel != null)
```

Comment:
I think, TelemetryClient is not s read-only property (please correct me if I am wrong).
If so, this is not a safe statement.
Please change to:

```
ITelemetryChannel telemetryChannel = this.configuration?.TelemetryChannel;
if (telemetryChannel != null)
{
    telemetryChannel.Flush();
}
```

Also - are you 100% sure that this is why the error was thrown and that id did not come out of the channel's Flush method?

Reply:
Sorry, typo: ... _TelemetryClient_ is not _a_ read-only property... :)

Reply:
Ok, agree with the change.

Re error  we got a NullReference exception on app shutdown (see linked issue in the [serilog repository](https://github.com/serilog/serilog-sinks-applicationinsights/issues/46)).
linke to [screenshot](https://user-images.githubusercontent.com/5837539/38416729-fa270aac-3995-11e8-8ddf-d4ef658d1377.png).
When I drilled into it, the channel was clearly null.
That happens when the configuration disposes, it sets it's channel property to null.

Reply:
Thanks. 

> Also - are you 100% sure that this is why the error was thrown and that id did not come out of the channel's Flush method?

I totally believe you that the error was due to the reason you describe and that the change fixed that reason. What I am saying is that there could also be another race on dispose inside the channel, and so even if your null check succeeds, telemetryChannel.Flush() might still throw from within Flush. It may also be a non-issue, just want to make sure that you thought of it.

Reply:
@macrogreg only looked at the `InMemoryChannel`, `transmitter` is read-only, shouldn't be null. And that `Flush` looks pretty guarded (try/catch at the end etc...).
Anyways, if we find more conditions, we can fix them later, too ;)
Letting NullReferences through to the consumer is never a good thing